Add account renaming functionality

enescakir opened the pull request:
# !! It doesn't have tests yet !!

- **Combine account password change and login pages**
  Previously, we had separate pages for changing passwords and logging in,
  which was unnecessary. This change combines them into a single unified
  page for a simpler user experience.
  
  I plan to add account renaming functionality to this page in a future
  commit.
  

- **Add path helper to account model**
  

- **Do not check authorization for renaming if it's passed as nil explicitly**
  We don't need to check authorization for all cases such as renaming
  accounts.
  

- **Skip storing audit logs when no project context is available**
  Some actions, such as renaming accounts, call the audit_log helper
  without a project context. In these cases, there's no relevant project
  information to log, so we now skip storing audit logs when the project
  context is missing.
  

- **Add account renaming functionality**
  Thanks to @jeremyevans's renaming helpers, it was straightforward to
  implement.
  
<img width="5120" height="2060" alt="account_rename" src="https://github.com/user-attachments/assets/e30c70bc-475a-4301-915c-68f0ef6d07d5" />

enescakir committed:
Add path helper to account model

enescakir committed:
Do not check authorization for renaming if it's passed as nil explicitly

We don't need to check authorization for all cases such as renaming
accounts.

enescakir committed:
Skip storing audit logs when no project context is available

Some actions, such as renaming accounts, call the audit_log helper
without a project context. In these cases, there's no relevant project
information to log, so we now skip storing audit logs when the project
context is missing.

enescakir committed:
Add account renaming functionality

Thanks to @jeremyevans's renaming helpers, it was straightforward to
implement.

enescakir commented:
Because of the first commit, authentication tests are failing with the following exception. @jeremyevans, how can I fix this issue? Architecturally, is the first commit incorrect?

```
8) Clover auth authenticated can change password when password entry is not required
     Failure/Error: Clover.app.call(env)

     NoMethodError:
       undefined method 'params' for #<Clover::RodaRequest:0x000000012a912eb0>
     # ./coverage/views/components-rodauth-login_field.erb.rb:14:in 'Tilt::CompiledTemplates#__tilt_2640'
     # ./coverage/views/account-settings.erb.rb:25:in 'block in Tilt::CompiledTemplates#__tilt_2640'
     # ./coverage/views/account-settings.erb.rb:19:in 'Tilt::CompiledTemplates#__tilt_2640'
     # ./routes/account.rb:15:in 'block (3 levels) in <class:Clover>'
     # ./routes/account.rb:11:in 'block (2 levels) in <class:Clover>'
     # ./routes/account.rb:5:in 'block in <class:Clover>'
     # ./clover.rb:1019:in 'block in <class:Clover>'
     # ./spec/routes/spec_helper.rb:64:in 'block (3 levels) in <top (required)>'
     # ./spec/routes/spec_helper.rb:63:in 'block (2 levels) in <top (required)>'
     # ./spec/routes/web/auth_spec.rb:447:in 'block (4 levels) in <top (required)>'
     # ./spec/spec_helper.rb:60:in 'block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:59:in 'block (2 levels) in <top (required)>'
```

jeremyevans commented:
> Because of the first commit, authentication tests are failing with the following exception. @jeremyevans, how can I fix this issue? Architecturally, is the first commit incorrect?

For catch use of untype-checked parameters in our templates, we undefine `params` during non-frozen testing. This breaks `rodauth.params`. This isn't usually a problem, because we don't undefine it until after `r.rodauth`. However, you are now loading a rodauth-specific template after `r.rodauth`.

To workaround this issue, you can change `rodauth.param('login')` to `typecast_params.str('login')` in the `login_field.erb` template (technically, `rodauth.param` returns an empty string instead of nil if the parameter is not submitted, but that shouldn't matter in this case).

jeremyevans commented on model/account.rb:
```diff
@@ -15,6 +15,10 @@ class Account < Sequel::Model(:accounts)
+  def path
+    "/account"
+  end
+
```
```suggestion
```

jeremyevans reviewed (commented):
I don't object to the combining of the change login and change login pages, but I don't think it makes things simpler. Changing login and changing password are generally done for different reasons, so putting them on the same page doesn't result in a simpler user experience to me. It also seems unrelated to account renaming. However, it doesn't cause harm.

enescakir committed:
Pass template instead of prefix to rename helper

We only use template_prefix in a single place, and the show_object
helper already accepts a template. This change makes them consistent and
also allows using the helper for resources without a show view.

enescakir committed:
Pass page to rename route helper

It allows to use the rename route helper other than in settings page.

enescakir committed:
Do not allow trailing spaces in account name

We allow spaces in account names and not allow leading spaces.

We should not allow trailing spaces too.
